### DISCUSSION TITLE
Welcome to taxonium Discussions!

### DISCUSSION THREAD

**Discussion by theosanderson** [theosanderson/taxonium#460] [Announcements]
## 👋 Welcome!
  We’re using Discussions as a place to connect with other members of our community. We hope that you:
  * Ask questions you’re wondering about.
  * Share ideas.
  * Engage with other community members.

**Comment by johnpage09** [comment]
Hi Theo,
I am trying to create a basic tree that I upload and run in Taxonium. The aim is to create a Taxonomy tree of various species in our database but unfortunately I am unable to get any results from the UI.

I have seen the tfci.jsonl but am unable to get the data model or the main fields that I need in order to get the visualisation to work.
Could you please help by supplying a file that has 10 nodes. It will help me understand the datamodel better.
Thanks,
John

**Reply by AngieHinrichs** [reply]
@johnpage09 have you already installed taxoniumtools?  (i.e. your Python environment setup of choice followed by `pip install -U taxoniumtools`)  Then you could create a simple Newick test tree with branch lengths and run `newick_to_taxonium` like this, to get a .jsonl.gz file with 10 leaves:
```
echo "((A:0, ((F:0, (H:0, (J:0):1):1):1, (G:0, (I:0):1):1):1):1, (B:0, (D:0, E:0):1):1, (C:0):1):0;" > test.nwk 
newick_to_taxonium  -i test.nwk -o test.jsonl.gz
```
`newick_to_taxonium` can also incorporate metadata from a TSV file -- run `newick_to_taxonium -h` to see options.

**Reply by theosanderson** [reply]
Thanks @AngieHinrichs!

@johnpage09 - to confirm, you have a couple of options

1. Simply use a conventional [Newick file](https://en.wikipedia.org/wiki/Newick_format) as the actual file you upload to taxonium.org
2.  As Angie says, create a special Taxonium JSONL file, in the format of `tfci.jsonl`, which you would normally do using the [newick_to_taxonium](https://docs.taxonium.org/en/latest/taxoniumtools.html#newick-to-taxonium) tool from from Taxoniumtools. This file contains the full layout of the tree so would be a hassle to generate in another way.

Please let us know if you continue to hit problems

**Reply by johnpage09** [reply]
Thank you both for getting back. I don't have a Newick Tree to start with. I am starting with an extract from a database that represents a  Taxonomy Tree. The idea is to get something similar to the Wikidata Visual Tree of Life.
This is my draft output that I am uploading onto the interface but the parsing appears to just hang.
Any tips would be much appreciated.
Thanks

[test_taxonium.zip](https://github.com/user-attachments/files/16975349/test_taxonium.zip)

On running it with the file, I get the following output 

![taxonium_output](https://github.com/user-attachments/assets/c540307d-3e0e-46f6-91b4-5a638a61c7b8)

**Reply by theosanderson** [reply]
Hi @johnpage09 , I haven't yet tried to look at the file you attached, but my advice would be that rather than trying to directly construct a JSONL file you instead try to make a Newick tree and a TSV metadata file. As I mentioned, those are supported by Taxonium, and they can also be converted to JSONL format using taxoniumtools. You will find it more intuitive to construct the Newick file than trying to reason about the Taxonium Jsonl

**Reply by johnpage09** [reply]
Thank you Theo. I will discuss the path you have indicated with my team. 
If you do get a chance to run my file, it would be much appreciated.
I have been unable to run a file with 5 to 10 nodes. Is there any restriction that the tree size has to have a certain number of nodes at a minimum ?

**Reply by theosanderson** [reply]
Hi @johnpage09 ,

I had a quick look at your file. I think that Taxonium is probably designed such that `node_ids` always have to go `0`, `1`, `2`, `3` etc. (and then parent_ids have to match them).

**Reply by johnpage09** [reply]
Very good. Thanks. I will make the change and try,

**Reply by johnpage09** [reply]
Hi Theo, 
I have made the changes suggested by you and checked the file for dangling pointers etc. It still hangs after 500 nodes and I can't see any errors in the Dev Console.
[test_taxonium_2.zip](https://github.com/user-attachments/files/16980842/test_taxonium_2.zip)
If you notice anything else in my file, please let me know.
Thanks

**Reply by theosanderson** [reply]
Hi John,

There is an error in the console
<img width="523" alt="image" src="https://github.com/user-attachments/assets/9aab8cc0-312d-4225-a2e2-1f9159f83491">

I believe the first node in the list (in the second line of the JSONL file) needs to be `0` then the next node needs to be `1`, etc. Really the `node_id` was just an identifier I added to help manual look up in the file

**Comment by johnpage09** [comment]
Thank you Theo. The error was not coming up in Chrome but I see it in Edge.
I have also fixed the part where I was missing node_id = parent_id. I will continue with the debugging and also look into your original suggestion.
Regards,
John

**Comment by johnpage09** [comment]
I think I have nearly got it working. However, I have another error related to the x_dist 
![image](https://github.com/user-attachments/assets/8f48c4ec-8b4a-4c13-8037-486123b7b8a7)
Could you please advise if there is a specific logic related to calculating these values.
My latest file is 
[test_taxonium_2.zip](https://github.com/user-attachments/files/16982181/test_taxonium_2.zip)
Many thanks

**Reply by theosanderson** [reply]
Your node ids go `1`, `2`, `3`, `4`, then `9`, `22`. They would need to be entirely sequential.

Sorry I likely won't be able to continue iterating on this much further. I would suggest you use the technique that Angie suggested to get yourself an example JSONL that works. If you continue to have problems I'd suggest using the Newick and TSV approach that I mentioned.

**Reply by johnpage09** [reply]
Thank you Theo.
